lorawan-device/src/region/fixed_channel_plans/join_channels.rs

```diff
@@ -64,7 +64,7 @@ impl JoinChannels {
                 self.num_retries += 1;
                 // pick a  random number 0-7 on the preferred subband
                 // NB: we don't use 500 kHz channels
-                let channel = (rng.next_u32() as u8 % 8) + ((sb as usize - 1) as u8 * 8);
+                let channel = ((rng.next_u32() & 0xFF) as u8 % 8) + ((sb as usize - 1) as u8 * 8);
```

Comment:
Not sure whether there's point need to clamp it as `32008 as u8` will be 8 anyway.
So this could be just simply `(rng.next_u32() % 8) as u8 + ...`